### DISCUSSION TITLE
Simplest example of LSP?

### DISCUSSION THREAD

**Discussion by tim-hardcastle** [microsoft/vscode-discussions#413] [Extension Development QnA] [answered]
I'm going to have to write an syntax-checking LSP soon and I'm finding it hard to grasp how it's all meant to work. What I'd like to see, if there is such a thing, is a completely minimal example, e.g. something like _all_ it does is if your code has the `.foo` extension it puts red wigglies under the word `foo` wherever it appears in your code. Ideally with the backend written in Go but that may be too much to hope for.

Is there such a thing? Thanks.

**Comment by isidorn** [comment, marked as answer]
Hi. There are a lot of lsp extension samples in this repository https://github.com/microsoft/vscode-extension-samples
I think that should help.

**Reply by tim-hardcastle** [reply]
Hi. I am aware of that repo but have no idea which part of it would contain the answer to my question. Instead of reading through all of it to find out what it does and if any of it is useful, I was hoping someone could point me to the one bit of information I actually need.

**Reply by tim-hardcastle** [reply]
This looks like what I need ... https://www.toptal.com/javascript/language-server-protocol-tutorial
